src/java/org/apache/ivy/core/cache/DefaultRepositoryCacheManager.java

```diff
@@ -1215,26 +1217,43 @@ public ResolvedResource resolve(Artifact artifact) {
                     for (int i = 0; i < confs.length; i++) {
                         Artifact[] arts = md.getArtifacts(confs[i]);
                         for (int j = 0; j < arts.length; j++) {
-                            Artifact transformedArtifact = NameSpaceHelper.transform(
-                                arts[j], options.getNamespace().getToSystemTransformer());
-                            ArtifactOrigin origin = getSavedArtifactOrigin(
-                                transformedArtifact);
-                            File artFile = getArchiveFileInCache(
-                                transformedArtifact, origin, false);
-                            if (artFile.exists()) {
-                                Message.debug("deleting " + artFile);
-                                if (!artFile.delete()) {
-                                    // Old artifacts couldn't get deleted!
-                                    // Restore the original ivy file so the next time we
-                                    // resolve the old artifacts are deleted again
-                                    backupDownloader.restore();
-                                    Message.error("Couldn't delete outdated artifact from cache: " + artFile);
-                                    return null;
-                                }
-                            }
-                            removeSavedArtifactOrigin(transformedArtifact);
+                            if (!prepAndDeleteArtifact(arts[j], options, backupDownloader)) return null;
                         }
                     }
+
+                    // sbt change, driven by https://github.com/sbt/sbt/issues/1750
+                    //
+                    // Typically Ivy behaviour, in the parsing above
+                    // (with ModuleDescriptorParser & PomModuleDescriptorBuilder),
+                    // involves making remote calls to locate sources and javadoc jars on the remote repository
+                    // and storing them in the ModuleDescriptor.
+                    //
+                    // However given a number of dependencies and resolver this suffers in performance,
+                    // particurlarly Ivy's lack of caching for not finding artifacts or IO failures.
+                    //
+                    // So sbt overrides this behaviour by failing fast (returning null).
+                    // See IvySbt.hasImplicitClassifier.
+                    //
+                    // This however has the side-effect that sources and javadoc jars aren't deleted here
+                    // so that they can be re-downloaded.
+                    //
+                    // So we keep the performance overrides and just delete those jars directly below.
```

Comment:
That PR is the + or - the equivalent of https://github.com/sbt/sbt/blob/v0.13.9/ivy/src/main/scala/sbt/ivyint/SbtChainResolver.scala#L67 (or at least with regards to the use case I've been looking at).

So if we merged that we would then have to programmatically enable that setting in sbt.

---

Despite there being two sbt-driven changes here, which together interact, their primary nature is different:

1) Don't always look for srcs/docs from remote you maniac
2) If we have a new descriptor then, in an effort to have a clean shop, let's always delete srcs/docs locally.

Given I'm not 100% that that PR would allow sbt to drop it's fail-fast, I'm leaning towards a "alwaysDeleteSrcsDocs" flag on `DefaultRepositoryCacheManager` provided sbt has access to enabling it somehow.

Or we could follow the lead on of that PR and set it in `ivySettings`.